pageserver/src/tenant/remote_timeline_client/index.rs

```diff
@@ -184,8 +193,67 @@ impl From<&LayerFileMetadata> for IndexLayerMetadata {
     }
 }
 
+/// Limited history of earlier ancestors.
+///
+/// A timeline can have more than 1 earlier ancestor, in the rare case that it was repeatedly
+/// reparented by having an later timeline be detached from it's ancestor.
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct Lineage {
+    /// Has the `reparenting_history` been truncated to `Lineage::REMEMBER_AT_MOST`.
+    #[serde(skip_serializing_if = "is_false", default)]
+    reparenting_history_truncated: bool,
+    /// Earlier ancestors, truncated when `reparented_overflown`
```

Comment:
`reparented_overflown` doesn't occur in this PR except here

Reply:
Good catch, thanks